src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,293 @@
+                            pmap_versions["carryover"] = True
+                            #pmap_versions["rebuild_gens"] = {}
+                    else:
+                        pmap_versions = {}
+                    self._pools[puuid][term] = {"rank": rank, "begin_time": datetime, "end_time": "", "host": host, "pid": pid, "logfile": fname, "maps": pmap_versions}
+                    #print(f"{datetime} FOUND pool {puuid} BEGIN\tterm {term} pmap_versions empty: {str(pmap_versions == {})} rank {rank}\t{host}\tPID {pid}\t{fname}")
+                    continue
+
+                # Find pool term end (PS leader step_down)
+                match = self.re_step_down.match(msg)
+                if match:
+                    puuid = match.group(1)
+                    term = int(match.group(2))
+                    if (term != self._cur_term[puuid]):
+                        print(f"WARN: step_down term={term} != cur_term={self._cur_term}")
+                    self._cur_ldr_rank[puuid] = -1
+                    self._cur_term[puuid] = -1
+                    self._pools[puuid][term]["end_time"] = datetime
+                    #print(f"{datetime} FOUND pool {puuid} END\tterm {term} rank {rank}\t{host}\tPID {pid}\t{fname}")
+                    continue
+
+                # TODO: find target status updates (precursor to pool map updates)
+
+                # Find pool map updates
+                # FIXME: but only on the current PS leader engine / term? And carry over latest pmap version from prior term to start with?
+                match = self.re_pmap_update.match(msg)
+                if match:
+                    puuid = match.group(1)
+                    from_ver = int(match.group(2))
+                    to_ver = int(match.group(3))
+                    # ignore if this engine is not the leader
+                    if puuid not in self._pools or rank != self._cur_ldr_rank[puuid]:
+                        continue
+                    term = self._cur_term[puuid]
+                    self._pools[puuid][term]["maps"][to_ver] = {"carryover": False, "from_ver": from_ver, "time": datetime, "rebuild_gens": {}}
+                    #print(f"FOUND pool {puuid} map update {from_ver}->{to_ver} rank {rank}\t{host}\tPID {pid}\t{fname}")
+                    continue
+
+                # Find rebuild start by the PS leader
+                match = self.re_rebuild_ldr_start.match(msg)
+                if match:
+                    puuid = match.group(1)
+                    mapver = int(match.group(2))
+                    rebuild_gen = int(match.group(3))
+                    rebuild_op = match.group(4)
+                    if rank != self._cur_ldr_rank[puuid]:
+                        continue
+                    term = self._cur_term[puuid]
+                    if term < 1:
+                        print(f"WARN pool {puuid} I don't know what term it is ({term})!")
```

Comment:
Depending on what tracks overall failure, might want to store warnings and errors somewhere instead of simply printing. Roughly e.g.

```
def _warn(self, message):
    print(f"WARN: {message}")
    self._warnings.append(message) # if you want to store the message
    self._warnings +=1 # if you just want a count
```

And then later can do
```
if self._warnings:
    print("Whoah something bad!")
    sys.exit(1) # or return 1, etc.
```